ui/segmented-bar.reel/segment.reel/segment.js

```diff
@@ -0,0 +1,93 @@
+/**
+ *@module ui/segmented-bar.reel/segment.reel
+ *@requires montage/ui/component
+ */
+var Component = require("montage/ui/component").Component;
+
+/**
+ * @class Segment
+ * @extends Component
+ * A Segment composes of data that forms a SegmentedBar. A Segment
+ * is bound to an object from SegmentedBar's content array. An array will be projected
+ * as a segment. The 'contentController' (rangeController/treeController)
+ * will use the proper logic and project the array of the rangeController or
+ * flatten the treeController data. The data projected will draw itself
+ * based on a percentage of the bar's total length. The remaining space
+ * will be its own Segment in the SegmentedBar.
+ */
+exports.Segment = Component.specialize({
+    label: {value: null},
+    _value: {value: null},
+    value: {
+        get: function () {return this._value},
+        set: function (value) {
+            this._value = value;
+            this._calculateLength(value);
+            this.needsDraw = true;
+        }
+    },
+    /**
+     * Length of each segment in %
+     * calculated based on data value & SegmentedBar's maxValue
+     */
+    _length: {value: null},
+    _calculateLength: {value: function _calculateLength(value) {
+        this._length = value / this.maxValue * 100 + "%";
+    }},
+
+    enterDocument: {value: function () {
+        // set initial segment width
+        this._calculateLength(this.value);
```

Comment:
Any reason to calculate the length each time the component is added to the Dom, or just the first time should be ok?